clippy_lints/src/long_variable_names.rs

```diff
@@ -0,0 +1,64 @@
+use clippy_config::Conf;
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_hir::{Pat, PatKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::impl_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for variable that exceeds a configurable number characters.
+    ///
+    /// ### Why is this bad?
+    /// Long variable names can make code harder to read and thus to maintain.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// let ferris_fixes_more_bugs_than_your_entire_devops_team_does = "content of a string";
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// let ferris_fixes_more_bugs = "content of a string";
+    /// ```
+    #[clippy::version = "1.88.0"]
+    pub LONG_VARIABLE_NAMES,
+    style,
```

Comment:
I think if we make it warn by default, we better pick a really large threshold. Adding warnings to many folks' CI jobs should have some serious benefit to make up for the hassle. So I'd either make it pedantic (which is allow by default) or set the default absurdly high, e.g. 100, so people who think this is a good idea can introduce tighter bounds to their projects.

Reply:
I agree that a default of 100 is reasonable and large enough. The pull request has been updated.